konduit-serving-core/src/main/java/ai/konduit/serving/launcher/command/ServeCommand.java

```diff
@@ -0,0 +1,236 @@
+            ExecUtils.addArgument(cmd, "-jar");
+            ExecUtils.addArgument(cmd, CommandLineUtils.getJar());
+        } else {
+            // probably a `vertx` command line usage, or in IDE.
+            ExecUtils.addArgument(cmd, CommandLineUtils.getFirstSegmentOfCommand());
+            ExecUtils.addArgument(cmd, "run");
+        }
+
+        cliArguments.forEach(arg -> ExecUtils.addArgument(cmd, arg));
+
+        try {
+            builder.command(cmd);
+            if (redirect) {
+                builder.redirectError(ProcessBuilder.Redirect.INHERIT);
+                builder.redirectOutput(ProcessBuilder.Redirect.INHERIT);
+            }
+            builder.start();
+            out.println(id);
+        } catch (Exception e) {
+            out.println("Cannot create konduit server process");
+            e.printStackTrace(out);
+            ExecUtils.exitBecauseOfProcessIssue();
+        }
+
+    }
+
+    private void addJavaCommand(List<String> cmd) {
+        if (ExecUtils.isWindows()) {
+            ExecUtils.addArgument(cmd, "cmd.exe");
+            ExecUtils.addArgument(cmd, "/C");
+            ExecUtils.addArgument(cmd, "start");
+            ExecUtils.addArgument(cmd, "serving-id - " + id);
+            ExecUtils.addArgument(cmd, "/B");
+        }
+        ExecUtils.addArgument(cmd, getJava().getAbsolutePath());
+
+        // Compute JVM Options
+        if (jvmOptions == null) {
+            String opts = System.getenv("JAVA_OPTS");
+            if (opts != null) {
+                Arrays.stream(opts.split(" ")).forEach(s -> ExecUtils.addArgument(cmd, s));
+            }
+        } else {
+            Arrays.stream(jvmOptions.split(" ")).forEach(s -> ExecUtils.addArgument(cmd, s));
+        }
+    }
+
+    private File getJava() {
+        File java;
+        File home = new File(System.getProperty("java.home"));
+        if (ExecUtils.isWindows()) {
+            java = new File(home, "bin/java.exe");
+        } else {
+            java = new File(home, "bin/java");
+        }
+
+        if (!java.isFile()) {
+            out.println("Cannot find java executable - " + java.getAbsolutePath() + " does not exist");
+            ExecUtils.exitBecauseOfSystemConfigurationIssue();
+        }
+        return java;
+    }
+
+    private boolean isLaunchedAsFatJar() {
+        return CommandLineUtils.getJar() != null;
+    }
+
+    private List<String> getArguments() {
+        List<String> args = executionContext.commandLine().allArguments();
+        // Add system properties passed as parameter
+        if (systemProperties != null) {
+            systemProperties.stream().map(entry -> "-D" + entry).forEach(args::add);
+        }
+
+        // Add id - it's important as it's the application mark.
+        args.add("-Dserving.id=" + getId());
+        return args;
+    }
+
+    private String getId() {
+        if (id == null) {
+            id = UUID.randomUUID().toString();
+        }
```

Comment:
Let's do something a little more user friendly than full UUIDs for default IDs, which end up looking like this:
```
20767bd7-0ef1-4164-80a4-b6391c727e69
```

Some options:
1. First 8 characters of UUID only
2. Millisecond timestamp (1585967431586 etc)
3. Human readable timestamp: YYYYMMDD-HHMMSS.sss
4. Some sort of "two word" name generator system like Docker has - https://github.com/moby/moby/blob/master/pkg/namesgenerator/names-generator.go

Maybe 1, but we revisit 4 later.

Reply:
Yup I'm gonna use JavaFaker at the end over here: https://github.com/DiUS/java-faker. Seems like a good solution for generation random names that's not already taken.

Reply:
Using the first 8 characters of UUID for now